sdk/go/common/util/cmdutil/profile.go

```diff
@@ -52,16 +54,43 @@ func CloseProfiling(prefix string) error {
 	pprof.StopCPUProfile()
 	trace.Stop()
 
+	// get up-to-date statistics
+	return writeMemoryProfile(prefix)
+}
+
+func writeMemoryProfile(prefix string) error {
 	mem, err := os.Create(fmt.Sprintf("%s.%v.mem", prefix, os.Getpid()))
 	if err != nil {
 		return fmt.Errorf("could not create memory profile: %w", err)
 	}
 	defer contract.IgnoreClose(mem)
 
-	runtime.GC() // get up-to-date statistics
+	runtime.GC()
 	if err = pprof.Lookup("allocs").WriteTo(mem, 0); err != nil {
 		return fmt.Errorf("could not write memory profile: %w", err)
 	}
-
 	return nil
 }
+
+func memoryProfileWriteLoop(prefix string) {
+	// Every second write a memory profile (in case we crash before we get a chance)
+	for i := 0; ; i++ {
+		// Sleep for a second
+		time.Sleep(5 * time.Second)
+
+		mem, err := os.Create(fmt.Sprintf("%s.%v.mem.%d", prefix, os.Getpid(), i))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "could not create memory profile: %v\n", err)
+			return
+		}
+
+		runtime.GC() // get up-to-date statistics
+		if err = pprof.WriteHeapProfile(mem); err != nil {
```

Comment:
The docs say 

> WriteHeapProfile is shorthand for [Lookup](https://pkg.go.dev/runtime/pprof#Lookup)("heap").WriteTo(w, 0). It is preserved for backwards compatibility.

https://pkg.go.dev/runtime/pprof#WriteHeapProfile

In `writeMemoryProfile` we write out the `allocs`, but there the `heap`, is that intended?

Reply:
Instead of changing it I'm going to just leave it consistent as is.  I am not sure which we should use.  One seems to get all the results of all allocations and the other the current state.  I am sure we chose allocs on purpose (all allocations).